client/executor/src/wasm_runtime.rs

```diff
@@ -185,11 +186,8 @@ impl RuntimeCache {
 		cache_path: Option<PathBuf>,
 		runtime_cache_size: u8,
 	) -> RuntimeCache {
-		RuntimeCache {
-			runtimes: Mutex::new(LruCache::new(runtime_cache_size.into())),
-			max_runtime_instances,
-			cache_path,
-		}
+		let cap = NonZeroUsize::new(runtime_cache_size as usize).expect("cache size is not zero");
+		RuntimeCache { runtimes: Mutex::new(LruCache::new(cap)), max_runtime_instances, cache_path }
```

Comment:
Honestly I'm not entirely sure if I'm a fan of this API change in `lru`, but I suppose this is fine as in practice you'll always want to have at least one runtime cached so that the performance doesn't completely tank.

Can you update the doc comment so that it says that `runtime_cache_size` must always be at least 1?

Also, can you modify `client/cli/src/commands/run_cmd.rs` and set a lower limit to `runtime_cache_size` so that the user can't specify anything lower than `1`?

I think something like this would work as an attribute `value_parser = clap::value_parser!(u8).range(1..))]`